src/llms/openai.ts

```diff
@@ -388,3 +390,153 @@ export const enabled = async (): Promise<OpenAIHealthDetails> => {
     { configured: details.openAI, ok: details.openAI } :
     details.openAI;
 }
+
+export enum StreamStatus {
+  IDLE = 'idle',
+  GENERATING = 'generating',
+  COMPLETED = 'completed',
+}
+
+export const TIMEOUT = 10000;
+
+
+export type OpenAIStreamState = {
+  setMessages: React.Dispatch<React.SetStateAction<Message[]>>;
+  reply: string;
+  streamStatus: StreamStatus;
+  error: Error | undefined;
+  value:
+    | {
+        enabled: boolean | undefined;
+        stream?: undefined;
+      }
+    | {
+        enabled: boolean | undefined;
+        stream: Subscription;
+      }
+    | undefined;
+}
```

Comment:
These need JSDoc comments on the types and fields 🙏

Reply:
Handled in latest commit